File: 0787-cheapest-flights-within-k-stops/0787-cheapest-flights-within-k-stops.cpp
class Solution {
public:
    int findCheapestPrice(int n, vector<vector<int>>& flights, int src, int dst, int k) {
        vector<pair<int,int>>adj[n];
        for(int i=0;i<flights.size();i++){
            adj[flights[i][0]].push_back({flights[i][1],flights[i][2]});
        }
    queue<pair<int,pair<int,int>>>q;
    // {stops,{src,cost}};
    q.push({0,{src,0}});
    // storing min cost for a particular node
    vector<int>cheapPriceDist(n,1e9);
    cheapPriceDist[src]=0;
    while(!q.empty()){
        auto it=q.front();
        q.pop();
        int stops=it.first;
        int node=it.second.first;
        int cost=it.second.second;
        if(stops>k)
         continue;

        for(auto i: adj[node]){
            int adjNode=i.first;
            int edgeWeight=i.second;
            if(cost+edgeWeight<cheapPriceDist[adjNode] && stops<=k){
                cheapPriceDist[adjNode]=cost+edgeWeight;
                q.push({stops+1,{adjNode,cost+edgeWeight}});
            }
        }
    }
    if(cheapPriceDist[dst]==1e9) return -1;
    return cheapPriceDist[dst];

    }
};

File: 0787-cheapest-flights-within-k-stops/0787-cheapest-flights-within-k-stops.js
/**
 * @param {number} n
 * @param {number[][]} flights
 * @param {number} src
 * @param {number} dst
 * @param {number} k
 * @return {number}
 */
var findCheapestPrice = function(n, flights, src, dst, k) {
      const adj= Array.from({length:n},()=>[]);
        for(let i=0;i<flights.length;i++){
            adj[flights[i][0]].push([flights[i][1],flights[i][2]]);
        }
    const q=[];
    // {stops,{src,cost}};
    q.push([0,src,0]);
    // storing min cost for a particular node
const cheapPriceDist=Array(n).fill(Infinity);
    cheapPriceDist[src]=0;
    while(q.length>0){
       let [stops,node,cost]=q.shift();
        if(stops>k)
         continue;
        
        for(const [adjNode,edgeWeight] of  adj[node]){
          
            if(cost+edgeWeight<cheapPriceDist[adjNode] && stops<=k){
                cheapPriceDist[adjNode]=cost+edgeWeight;
                q.push([stops+1,adjNode,cost+edgeWeight]);
            }
        }
    }
    if(cheapPriceDist[dst]==Infinity) return -1;
    return cheapPriceDist[dst];

};